execution_engine_testing/tests/src/test/explorer/faucet.rs

```diff
@@ -1,77 +1,1210 @@
+fn should_allow_installer_to_set_variables() {
+    let installer_secret_key =
+        SecretKey::ed25519_from_bytes([1; 32]).expect("failed to create secret key");
+    let installer_public_key = PublicKey::from(&installer_secret_key);
+    let installer_account = AccountHash::from_public_key(&installer_public_key, crypto::blake2b);
+
     let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&DEFAULT_RUN_GENESIS_REQUEST).commit();
+
+    let fund_installer_account_request = ExecuteRequestBuilder::transfer(
+        *DEFAULT_ACCOUNT_ADDR,
+        runtime_args! {
+            mint::ARG_TARGET => installer_account,
+            mint::ARG_AMOUNT => 300_000_000_000_000u64,
+            mint::ARG_ID => <Option<u64>>::None
+        },
+    )
+    .build();
+
+    builder
+        .exec(fund_installer_account_request)
+        .expect_success()
+        .commit();
+
+    let installer_session_request = ExecuteRequestBuilder::standard(
+        installer_account,
+        FAUCET_INSTALLER_SESSION,
+        runtime_args! {ARG_ID => 1337u64, ARG_AMOUNT => U512::from(500_000u64)},
+    )
+    .build();
+
+    builder
+        .exec(installer_session_request)
+        .expect_success()
+        .commit();
+
+    let faucet_contract_hash = builder
+        .get_expected_account(installer_account)
+        .named_keys()
+        .get(FAUCET_CONTRACT_NAMED_KEY)
+        .cloned()
+        .and_then(Key::into_hash)
+        .map(ContractHash::new)
+        .expect("failed to find faucet contract");
+
+    // check the balance of the faucet's purse before
+    let faucet_contract = builder
+        .get_contract(faucet_contract_hash)
+        .expect("failed to find faucet contract");
+
+    let faucet_purse = faucet_contract
+        .named_keys()
+        .get(FAUCET_PURSE_NAMED_KEY)
+        .cloned()
+        .and_then(Key::into_uref)
+        .expect("failed to find faucet purse");
+
+    let faucet_purse_balance = builder.get_purse_balance(faucet_purse);
+    assert_eq!(faucet_purse_balance, U512::from(500_000u64));
+
+    let available_amount = builder
+        .query(
+            None,
+            faucet_contract_hash.into(),
+            &[AVAILABLE_AMOUNT_NAMED_KEY.to_string()],
+        )
+        .expect("failed to find available amount named key")
+        .as_cl_value()
+        .cloned()
+        .expect("failed to convert to cl value")
+        .into_t::<U512>()
+        .expect("failed to convert into U512");
+
+    // the available amount per interval will be zero until the installer calls
+    // the set_variable entrypoint to finish setup.
+    assert_eq!(available_amount, U512::zero());
+
+    let time_interval = builder
+        .query(
+            None,
+            faucet_contract_hash.into(),
+            &[TIME_INTERVAL_NAMED_KEY.to_string()],
+        )
+        .expect("failed to find time interval named key")
+        .as_cl_value()
+        .cloned()
+        .expect("failed to conert to cl value")
```

Comment:
```suggestion
        .expect("failed to convert to cl value")
```
There are another few similar typos - suggest you find and replace `conert`.